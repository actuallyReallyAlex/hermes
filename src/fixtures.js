export const defaultState = {
  ship: {
    cargo: {
      items: [],
      volumeRemaining: 5
    },
    location: {
      name: null,
      value: null
    }
  },
  ui: {
    view: 'Ship'
  },
  user: {
    cash: 100
  },
  world: {
    isTimerRunning: false,
    planets: [
      {
        id: "0",
        isHomePlanet: true,
        location: 0,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '0',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '1',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '2',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '3',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '4',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 1'
      },
      {
        id: "1",
        isHomePlanet: false,
        location: 50,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '5',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '6',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '7',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '8',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '9',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          }
        ],
        name: 'Test Planet 2'
      },
      {
        id: "2",
        isHomePlanet: false,
        location: 100,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '10',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '11',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '12',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '13',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '14',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 3'
      }
    ]
  }
}
