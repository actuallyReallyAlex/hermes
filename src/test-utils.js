import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { defaultState } from './fixtures'
import { render } from '@testing-library/react'

const mockThunks = ({ dispatch, getState }) => next => action => {
  const mockedAction = () => ({ type: 'MOCKED ACTION' })

  let returnValue

  if (typeof action === 'function' || action === undefined) {
    returnValue = mockedAction
  } else {
    returnValue = next(action)
  }

  return returnValue
}

const middlewares = [mockThunks]

export const mockStore = configureStore(middlewares)

export const customRender = options => {
  const { component, state, props } = options

  const Component = component

  return render(
    <Provider store={mockStore(state ? state : defaultState)}>
      <Component {...props} />
    </Provider>
  )
}
