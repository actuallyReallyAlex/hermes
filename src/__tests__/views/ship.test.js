import { customRender } from '../../test-utils'
import ShipView from '../../views/ship'
import { defaultState } from '../../fixtures'
import { waitForElement, fireEvent } from '@testing-library/dom'

const customState = {
  ...defaultState,
  ship: {
    ...defaultState.ship,
    cargo: {
      items: [
        {
          destination: {
            name: 'Test Planet 2',
            value: 50
          },
          id: '0',
          name: 'Test Item',
          quantity: 1,
          volume: 1,
          value: 1
        }
      ],
      volumeRemaining: 4
    }
  }
}

describe('<ShipView />', () => {
  it('Should render the <ShipView /> component.', () => {
    const container = customRender({ component: ShipView, state: customState })
    expect(container.asFragment()).toMatchSnapshot()
  })

  it('Should handle removing an item from ship cargo.', async () => {
    const { getByTestId } = customRender({
      component: ShipView,
      state: customState
    })
    const button = await waitForElement(() => getByTestId('remove-button-0'))
    fireEvent.click(button)
  })
})
