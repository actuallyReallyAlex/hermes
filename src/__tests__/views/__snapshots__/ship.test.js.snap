// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShipView /> Should render the <ShipView /> component. 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Your Ship
    </h2>
    <h3>
      Cargo
    </h3>
    <div
      class="StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <span
        class="StyledText-sc-1sadyjn-0 bnqisW"
      >
        Volume Remaining:
      </span>
      <span
        class="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        4
      </span>
    </div>
    <div
      class="Table_tableWrapper__1fXsJ Table_tableScroll__zRwCt undefined"
    >
      <table
        class="Table_table__1DoSj"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Destination
            </th>
            <th>
              Quantity
            </th>
            <th>
              Remove
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="sc-bdVaJa iJTrNS"
          >
            <td>
              Test Item
            </td>
            <td>
              Test Planet 2
            </td>
            <td>
              1
            </td>
            <td>
              <button
                class="StyledButton-sc-323bzc-0 jAPRVi"
                data-testid="remove-button-0"
                type="button"
              >
                <svg
                  aria-label="Subtract"
                  class="StyledIcon-ofa7kd-0 iOkQrb"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2,12 L22,12"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <h3>
      Location:
    </h3>
    <div
      class="StyledBox-sc-13pk1d4-0 MmWcm"
    >
      <span
        class="StyledText-sc-1sadyjn-0 isVaoK"
      >
        Value:
      </span>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
      />
      <span
        class="StyledText-sc-1sadyjn-0 gNZQxf"
      />
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
      />
      <span
        class="StyledText-sc-1sadyjn-0 isVaoK"
      >
        Name:
      </span>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
      />
      <span
        class="StyledText-sc-1sadyjn-0 gNZQxf"
      />
    </div>
  </div>
</DocumentFragment>
`;
